secure/key-manager/src/tests.rs

```diff
@@ -531,47 +586,106 @@ fn verify_init_and_basic_rotation<T: LibraInterface>(mut node: Node<T>) {
 }
 
 #[test]
-// This tests the application's main loop to ensure it handles basic operations and reliabilities
-fn test_main_loop() {
+// This tests the application's main loop to ensure it handles basic operations and reliabilities.
+// To do this, the test repeatedly calls "execute_once_and_sleep" -- identical to the main "execute"
+// loop.
+fn test_execute() {
     // Test the mock libra interface implementation
     let (config, _genesis_key) = config_builder::test_config();
-    let node = setup_node_using_test_mocks(&config);
-    verify_main_loop(node);
+    let mut node = setup_node_using_test_mocks(&config);
+    node.key_manager.libra.mock_libra_timestamp();
+    verify_execute(node);
 
     // Test the json libra interface implementation
     let (config, _genesis_key) = config_builder::test_config();
-    let (node, _runtime) = setup_node_using_json_rpc(&config);
-    verify_main_loop(node);
+    let (mut node, _runtime) = setup_node_using_json_rpc(&config);
+    node.key_manager.libra.mock_libra_timestamp();
+    verify_execute(node);
 }
 
-fn verify_main_loop<T: LibraInterface>(mut node: Node<T>) {
-    // Verify nothing to be done initially
-    let mut action = node.key_manager.evaluate_status().unwrap();
-    assert_eq!(action, Action::NoAction);
-    node.key_manager.perform_action(action).unwrap();
+fn verify_execute<T: LibraInterface>(mut node: Node<T>) {
+    // Verify correct initial state (i.e., nothing to be done by key manager)
+    assert_eq!(0, node.time.now());
+    assert_eq!(0, node.libra.last_reconfiguration().unwrap());
+
+    // Perform a single execution iteration and verify sleep elapsed
+    assert!(node.key_manager.execute_once_then_sleep().is_ok());
+    assert_eq!(crate::SLEEP_PERIOD_SECS, node.time.now());
```

Comment:
should avoid testing "sleeping" in tests unless this is a virtual sleep

Reply:
I guess it might be

Reply:
Yes, it is a virtual sleep, not a real sleep, but I've removed this anyway seeing as though we've removed the sleep() from the execute_once() call.